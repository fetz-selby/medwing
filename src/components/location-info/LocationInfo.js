import React from 'react';
import PropTypes from 'prop-types';
import wrong from '../../assets/icons/wrong.svg';
import './location-info.css';

const LocationInfo = (props)=>
    <div className='location-info-container'>
        <div className='icon-layout'>
            <img className='icon' src={wrong} alt='no location' />
        </div>
        <div className='info-layout'>
            <div className='message-container'>
                <div className='message'>{props.message}</div>
            </div>
            <div className='action-container'>
                <div className='btn-link' onClick={props.onClose}>close</div>
            </div>
        </div>
    </div>


LocationInfo.propTypes = {
    message: PropTypes.string.isRequired,
    onClose: PropTypes.func
}

export default LocationInfo;