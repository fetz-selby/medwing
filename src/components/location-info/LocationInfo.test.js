import React from 'react';
import {shallow} from 'enzyme';
import LocationInfo from './';

describe('Location Info Component', ()=>{
    it('should display [message] passed with props', ()=>{
        const message = 'location could not be found';

        const locationInfo = shallow(<LocationInfo message={message}/>);
        expect(locationInfo.find('.message').text()).toBe(message);
    })

    it('should call onClose when close is clicked', ()=>{
        const message = 'location could not be found';
        const id = 1;
        const onClose = jest.fn();

        const props = {
            message,
            onClose
        }

        const locationInfo = shallow(<LocationInfo {...props}/>);
        locationInfo.find('.btn-link').simulate('click');
        expect(onClose).toHaveBeenCalled();
    })

    it('should render correctly', ()=>{
        const message = 'location could not be found';

        const props = {
            message
        }

        const locationInfo = shallow(<LocationInfo {...props}/>);
        expect(locationInfo).toMatchSnapshot();
    })
})