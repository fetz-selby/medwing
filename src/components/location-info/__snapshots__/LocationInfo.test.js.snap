// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Info Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationInfo
    message="location could not be found"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icon-layout"
        >
          <img
            alt="no location"
            className="icon"
            src="wrong.svg"
          />
        </div>,
        <div
          className="info-layout"
        >
          <div
            className="message-container"
          >
            <div
              className="message"
            >
              location could not be found
            </div>
          </div>
          <div
            className="action-container"
          >
            <div
              className="btn-link"
            >
              close
            </div>
          </div>
        </div>,
      ],
      "className": "location-info-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="no location"
            className="icon"
            src="wrong.svg"
          />,
          "className": "icon-layout",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "no location",
            "className": "icon",
            "src": "wrong.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="message-container"
            >
              <div
                className="message"
              >
                location could not be found
              </div>
            </div>,
            <div
              className="action-container"
            >
              <div
                className="btn-link"
              >
                close
              </div>
            </div>,
          ],
          "className": "info-layout",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="message"
              >
                location could not be found
              </div>,
              "className": "message-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "location could not be found",
                "className": "message",
              },
              "ref": null,
              "rendered": "location could not be found",
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="btn-link"
              >
                close
              </div>,
              "className": "action-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "close",
                "className": "btn-link",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "close",
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon-layout"
          >
            <img
              alt="no location"
              className="icon"
              src="wrong.svg"
            />
          </div>,
          <div
            className="info-layout"
          >
            <div
              className="message-container"
            >
              <div
                className="message"
              >
                location could not be found
              </div>
            </div>
            <div
              className="action-container"
            >
              <div
                className="btn-link"
              >
                close
              </div>
            </div>
          </div>,
        ],
        "className": "location-info-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="no location"
              className="icon"
              src="wrong.svg"
            />,
            "className": "icon-layout",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "no location",
              "className": "icon",
              "src": "wrong.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="message-container"
              >
                <div
                  className="message"
                >
                  location could not be found
                </div>
              </div>,
              <div
                className="action-container"
              >
                <div
                  className="btn-link"
                >
                  close
                </div>
              </div>,
            ],
            "className": "info-layout",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="message"
                >
                  location could not be found
                </div>,
                "className": "message-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "location could not be found",
                  "className": "message",
                },
                "ref": null,
                "rendered": "location could not be found",
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="btn-link"
                >
                  close
                </div>,
                "className": "action-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "close",
                  "className": "btn-link",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "close",
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
